cmd/internal/runcmd/runcmd.go

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package runcmd provides support for running Cobra commands in main functions.
+package runcmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	"go.uber.org/dig"
+)
+
+// Run executes a cobra command and handles the results.  It is intended
+// for use in `main` functions, supplying the necessary error-handling and
+// exiting the process with an appropriate status.
+//
+// This function does not return.
+func Run(cmd *cobra.Command) {
+	// always silence errors, since they are handled here
+	cmd.SilenceErrors = true
+
+	err := cmd.Execute()
+	if err != nil {
+		displayError(err, cmd.ErrOrStderr())
+		os.Exit(-1)
```

Comment:
If you do the `exit()` here, I'm assuming that testing the `Run` gets tricky. Can we move the `exit()` back into the caller and test the `Run()` instead of just `displayError`?

Reply:
That's why I moved the "interesting" bits to displayError, so they could be tested.  I think the remainder is pretty much what would be in `main` otherwise, so equivalently difficult to test in either case (and easy enough to test by hand, since it's only got one branch).

Reply:
I see your point though anything that depends on manual testing would be good to automate. Even though we only have simple logic here, it's probably good to test the method that will be called (`Run()`) instead of the helper (`displayError()`).

Reply:
How do you suggest doing that?

Reply:
We could return an error in `Run` to the caller (`func Run(cmd *cobra.Command) error`) and caller would be responsible for the exit. If we do that, we should be able to test that conditional with a `&cobra.Command{}` stub.